### PR TITLE
Configure Gitea registration preferences

### PR DESCRIPTION
**Pull request by anatbal**
# PR for issue #348 

## What is being addressed

Configure Gitea registration preferences

## How is this addressed
- Block normal users registrations by GITEA__service__DISABLE_REGISTRATION in the gitea config
- Add an admin user in order for the TRE admin to be able to push git repositories

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 SHELL:=/bin/bash
 
-all: bootstrap mgmt-deploy build-api-image push-api-image build-resource-processor-vm-porter-image push-resource-processor-vm-porter-image tre-deploy
+all: bootstrap mgmt-deploy build-api-image push-api-image build-resource-processor-vm-porter-image push-resource-processor-vm-porter-image push-gitea-image tre-deploy
```
Comment (marrobi) [inline Makefile:5]: For the other images we have a build and push stage - might use build if want to test locally without push - can we do the same for gitea?

File: templates/core/terraform/main.tf [hunk 2]
```diff
@@ -255,6 +255,10 @@ module "gitea" {
   source   = "../../shared_services/gitea/terraform"
   tre_id   = var.tre_id
   location = var.location
+  management_api_image_tag =  var.management_api_image_tag
```
Comment (marrobi) [inline templates/core/terraform/main.tf:258]: nit - formatting (prevents churn later)

File: templates/shared_services/gitea/terraform/gitea-webapp.tf [hunk 3]
```diff
@@ -10,8 +18,15 @@ resource "azurerm_app_service" "gitea" {
...
-    TRE_ID            = var.tre_id
-    RESOURCE_LOCATION = var.location
+     # Settings for private Container Registires  
+    "DOCKER_REGISTRY_SERVER_USERNAME"            = var.docker_registry_username
+    "DOCKER_REGISTRY_SERVER_URL"                 = "https://${var.docker_registry_server}"
+    "DOCKER_REGISTRY_SERVER_PASSWORD"            = var.docker_registry_password
+   
+    # TBD, Username and password should not be defined here: #542
+    GITEA_USERNAME                 = "giteaadmin"
+    GITEA_PASSWD                   = random_password.gitea_passwd.result
```
Comment (marrobi) [inline templates/shared_services/gitea/terraform/gitea-webapp.tf:28]: Can we save the password in keyvault, as per https://github.com/microsoft/AzureTRE/blob/b80eb68503e5b2f4f5368f0a36e247ec6b3ffb6b/templates/core/terraform/resource_processor/vmss_porter/main.tf#L40

**Review by marrobi — APPROVED** [review]
Looks good, couple of suggestions that would be good to address as part of this PR or later.

File: templates/shared_services/gitea/terraform/gitea-webapp.tf (hunk 3 above)
Comment (anatbal) [reply templates/shared_services/gitea/terraform/gitea-webapp.tf:28]: @marrobi Sure, we are working on it as part of a different task (#542 ) just in order to make it in a separate PR

File: Makefile (hunk 1 above)
Comment (anatbal) [reply Makefile]: Fixed.

File: templates/core/terraform/main.tf (hunk 2 above)
Comment (anatbal) [reply templates/core/terraform/main.tf]: Fixed.

File: templates/core/terraform/main.tf [hunk 4]
```diff
@@ -221,10 +221,13 @@ module "bastion" {
...
 module "gitea" {
   count = var.deploy_gitea == true ? 1 : 0
-
-  source   = "../../shared_services/gitea/terraform"
-  tre_id   = var.tre_id
-  location = var.location
+  source                                     = "../../shared_services/gitea/terraform"
+  tre_id                                     = var.tre_id
+  location                                   = var.location
+  management_api_image_tag                   = var.management_api_image_tag
```
Comment (tamirkamara) [inline templates/core/terraform/main.tf:227]: need to change this "management_api" prefix. I think we can just remove it.

**Review by tamirkamara — CHANGES_REQUESTED** [review]

**Review by tamirkamara — APPROVED** [review]
LGTM

**Pull request merged**